File: microservices/PassengerService/src/main/java/com/modsen/taxi/passengerservice/aspect/LoggingAspect.java
package com.modsen.taxi.passengerservice.aspect;

import lombok.extern.slf4j.Slf4j;
import org.aspectj.lang.JoinPoint;
import org.aspectj.lang.annotation.AfterReturning;
import org.aspectj.lang.annotation.AfterThrowing;
import org.aspectj.lang.annotation.Aspect;
import org.aspectj.lang.annotation.Before;
import org.springframework.stereotype.Component;

import java.util.Arrays;


@Aspect
@Component
@Slf4j
public class LoggingAspect {

    @Before("execution(* com.modsen.taxi.passengerservice.service.*.*(..))")
    public void logBefore(JoinPoint joinPoint) {
        log.info("Entering method: {} with arguments: {}",
                joinPoint.getSignature(),
                Arrays.toString(joinPoint.getArgs())
        );
    }


    @AfterReturning(
            pointcut = "execution(* com.modsen.taxi.passengerservice.service.*.*(..))",
            returning = "result"
    )
    public void logAfterReturning(JoinPoint joinPoint, Object result) {
        log.info("Exiting method: {} with result: {}",
                joinPoint.getSignature(),
                result
        );
    }


    @AfterThrowing(
            pointcut = "execution(* com.modsen.taxi.passengerservice.service.*.*(..))",
            throwing = "exception"
    )
    public void logAfterThrowing(JoinPoint joinPoint, Throwable exception) {
        log.error("Exception in method: {} with message: {}",
                joinPoint.getSignature(),
                exception.getMessage()
        );
    }
}

File: microservices/PassengerService/src/main/java/com/modsen/taxi/passengerservice/aspect/LoggingAspect.kt
package com.modsen.taxi.passengerservice.aspect

import org.aspectj.lang.JoinPoint
import org.aspectj.lang.annotation.AfterReturning
import org.aspectj.lang.annotation.AfterThrowing
import org.aspectj.lang.annotation.Aspect
import org.aspectj.lang.annotation.Before
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Aspect
@Component
open class LoggingAspect {

    private val logger: Logger = LoggerFactory.getLogger(LoggingAspect::class.java)

    @Before("execution(* com.modsen.taxi.passengerservice.service.*.*(..))")
    fun logBefore(joinPoint: JoinPoint) {
        logger.info("Entering method: {} with arguments: {}",
            joinPoint.signature,
            joinPoint.args.joinToString(", ")
        )
    }

    @AfterReturning(
        pointcut = "execution(* com.modsen.taxi.passengerservice.service.*.*(..))",
        returning = "result"
    )
    fun logAfterReturning(joinPoint: JoinPoint, result: Any?) {
        logger.info("Exiting method: {} with result: {}",
            joinPoint.signature,
            result
        )
    }

    @AfterThrowing(
        pointcut = "execution(* com.modsen.taxi.passengerservice.service.*.*(..))",
        throwing = "exception"
    )
    fun logAfterThrowing(joinPoint: JoinPoint, exception: Throwable) {
        logger.error("Exception in method: {} with message: {}",
            joinPoint.signature,
            exception.message
        )
    }
}
